persistence/src/main/java/fi/otavanopisto/pyramus/domainmodel/matriculation/MatriculationExamAttendance.java

```diff
@@ -0,0 +1,162 @@
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.TableGenerator;
+import javax.persistence.Version;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+
+import fi.otavanopisto.pyramus.domainmodel.base.ArchivableEntity;
+
+@Entity
+@Indexed
+public class MatriculationExamAttendance implements ArchivableEntity {
+
+  /**
+   * Returns the unique identifier of this object.
+   * 
+   * @return The unique identifier of this object
+   */
+  public Long getId() {
+    return id;
+  }
+
+  @Override
+  public Boolean getArchived() {
+    return archived;
+  }
+
+  @Override
+  public void setArchived(Boolean archived) {
+    this.archived = archived;
+  }
+  
+  public Long getVersion() {
+    return version;
+  }
+
+  public void setVersion(Long version) {
+    this.version = version;
+  }
+
+  public void setId(Long id) {
+    this.id = id;
+  }
+  
+  public MatriculationExamEnrollment getEnrollment() {
+    return enrollment;
+  }
+
+  public void setEnrollment(MatriculationExamEnrollment enrollment) {
+    this.enrollment = enrollment;
+  }
+
+  public MatriculationExamSubject getSubject() {
+    return subject;
+  }
+
+  public void setSubject(MatriculationExamSubject subject) {
+    this.subject = subject;
+  }
+
+  public Boolean isMandatory() {
+    return mandatory;
+  }
+
+  public void setMandatory(Boolean mandatory) {
+    this.mandatory = mandatory;
+  }
+
+  public Boolean isRepeat() {
+    return repeat;
+  }
+
+  public void setRepeat(Boolean repeat) {
+    this.repeat = repeat;
+  }
+
+  public Integer getYear() {
+    return year;
+  }
+
+  public void setYear(Integer year) {
+    this.year = year;
+  }
+
+  public MatriculationExamTerm getTerm() {
+    return term;
+  }
+
+  public void setTerm(MatriculationExamTerm term) {
+    this.term = term;
+  }
+
+  public MatriculationExamAttendanceStatus getStatus() {
+    return status;
+  }
+
+  public void setStatus(MatriculationExamAttendanceStatus status) {
+    this.status = status;
+  }
+
+  public MatriculationExamGrade getGrade() {
+    return grade;
+  }
+
+  public void setGrade(MatriculationExamGrade grade) {
+    this.grade = grade;
+  }
+
+  @Id
+  @GeneratedValue(strategy=GenerationType.TABLE, generator="MatriculationExamAttendance")  
+  @TableGenerator(name="MatriculationExamAttendance", allocationSize=1, table = "hibernate_sequences", pkColumnName = "sequence_name", valueColumnName = "sequence_next_hi_value")
+  @DocumentId 
+  private Long id;
+  
+  @ManyToOne
+  private MatriculationExamEnrollment enrollment;
+  
+  @Column
+  @Enumerated(EnumType.STRING)
+  private MatriculationExamSubject subject;
+  
+  @Column
+  private Boolean mandatory;
+  
+  @Column(name = "repeat_")
```

Comment:
Database script has just 'repeat', is there an issue here? Also something like reattempt or retry might be better suited term (for alternatives to using _ in the name as it is sometimes treated as wildcard by db).

Reply:
Renamed field